adapters/trafficgate/trafficgate.go

```diff
@@ -0,0 +1,187 @@
+package trafficgate
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"text/template"
+
+	"github.com/mxmCherry/openrtb/v15/openrtb2"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/macros"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+type adapter struct {
+	EndpointTemplate *template.Template
+}
+
+type BidResponseExt struct {
+	Prebid struct {
+		Type string
+	}
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, reqInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+
+	errs := make([]error, 0, len(request.Imp))
+	headers := http.Header{
+		"Content-Type": {"application/json"},
+		"Accept":       {"application/json"},
+	}
+
+	// Pull the host and source ID info from the bidder params.
+	reqImps, err := splitImpressions(request.Imp)
+
+	if err != nil {
+		errs = append(errs, err)
+	}
```

Comment:
Since `splitImpressions` will return a nil array and an error if it encounters a problem, I recommend exiting out early here. Something like:
```
reqImps, err := splitImpressions(request.Imp)
if err != nil {
	return nil, []error{err}
}
```

Reply:
Done